packages/pipeline/src/utils/index.ts

```diff
@@ -51,3 +51,14 @@ export function handleError(e: any): void {
     }
     process.exit(1);
 }
+
+/**
+ * Does fetchAsync(), and checks the status code, throwing if it doesn't indicate success.
```

Comment:
Optional: Let's clarify that "success" in this context means a status code between 200 and 299.

https://developer.mozilla.org/en-US/docs/Web/API/Response/ok